### PR TITLE
[vf, jsp] Adapt JSP & VF modules to new javacc wrapper

### PR DESCRIPTION
**Pull request by oowekyala**
Part of #2239

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/cpd/VfTokenizer.java [hunk 1]
```diff
@@ -4,42 +4,19 @@
...
-                tokenEntries.add(new TokenEntry(String.valueOf(currentToken.kind), sourceCode.getFileName(),
-                        currentToken.beginLine, currentToken.beginColumn, currentToken.endColumn));
-                currentToken = (Token) tokenMgr.getNextToken();
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        tokenEntries.add(TokenEntry.getEOF());
+    protected TokenManager getLexerForSource(SourceCode sourceCode) {
+        return new VfTokenManager(new CharSequenceReader(sourceCode.getCodeBuffer()));
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/cpd/VfTokenizer.java]: We need to skip the BOM to preserve same behavior...

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTAttribute.java [hunk 2]
```diff
@@ -5,18 +5,14 @@
 
 package net.sourceforge.pmd.lang.vf.ast;
 
-public class ASTAttribute extends AbstractVFNode {
+public class ASTAttribute extends AbstractVfNode {
 
     private String name;
 
-    public ASTAttribute(int id) {
+    ASTAttribute(int id) {
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTAttribute.java]: The setter "setName" can be package-private as well

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDeclaration.java [hunk 3]
```diff
@@ -5,18 +5,14 @@
 
 package net.sourceforge.pmd.lang.vf.ast;
 
-public class ASTDeclaration extends AbstractVFNode {
+public class ASTDeclaration extends AbstractVfNode {
 
     private String name;
 
-    public ASTDeclaration(int id) {
+    ASTDeclaration(int id) {
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDeclaration.java]: The setter "setName" can be package-private as well.

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDoctypeDeclaration.java [hunk 4]
```diff
@@ -5,21 +5,17 @@
...
-public class ASTDoctypeDeclaration extends AbstractVFNode {
+public class ASTDoctypeDeclaration extends AbstractVfNode {
 
     /**
      * Name of the document type. Cannot be null.
      */
     private String name;
 
-    public ASTDoctypeDeclaration(int id) {
+    ASTDoctypeDeclaration(int id) {
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDoctypeDeclaration.java]: The setter "setName" can be package-private as well.

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDoctypeExternalId.java [hunk 5]
```diff
@@ -17,14 +17,10 @@
      */
     private String publicId;
 
-    public ASTDoctypeExternalId(int id) {
+    ASTDoctypeExternalId(int id) {
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTDoctypeExternalId.java]: The setters "setPublicId" and "setUri" can be package-private

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTElement.java [hunk 6]
```diff
@@ -23,14 +23,10 @@
      */
     private boolean unclosed;
 
-    public ASTElement(int id) {
+    ASTElement(int id) {
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/ASTElement.java]: The setters "setEmpty", "setName", "setUnclosed" can be package-private

File: pmd-jsp/src/main/java/net/sourceforge/pmd/cpd/JSPTokenizer.java [hunk 7]
```diff
@@ -4,38 +4,17 @@
...
-                tokenEntries.add(new TokenEntry(String.valueOf(currentToken.kind), sourceCode.getFileName(),
-                        currentToken.beginLine, currentToken.beginColumn, currentToken.endColumn));
-                currentToken = (Token) tokenMgr.getNextToken();
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        tokenEntries.add(TokenEntry.getEOF());
+    protected TokenManager getLexerForSource(SourceCode sourceCode) {
+        return new JspTokenManager(new CharSequenceReader(sourceCode.getCodeBuffer()));
```
[the remark this answers is not in the document]
Comment (oowekyala) [reply pmd-jsp/src/main/java/net/sourceforge/pmd/cpd/JSPTokenizer.java]: Can we just use `IOUtil::skipBOM`?

File: pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/VfTokenManager.java [hunk 8]
```diff
@@ -23,10 +23,10 @@
 @Deprecated
 @InternalApi
```
Comment (adangel) [inline pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/VfTokenManager.java]: Not sure, whether we decided yet: Are the `*TokenManager` classes internal api or not? The JavaTokenManager isn't...

Comment (adangel) [reply pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/VfTokenManager.java]: I'll remove `@Deprecated` and keep `@InternalApi` for now...

**Pull request merged**